services/provider/server/server.go

```diff
@@ -161,22 +188,17 @@ func (s *OCSProviderServer) OnboardConsumer(ctx context.Context, req *pb.Onboard
 		return nil, status.Error(codes.InvalidArgument, err.Error())
 	}
 
-	storageConsumerUUID, err := s.consumerManager.Create(ctx, req, int(storageQuotaInGiB))
+	storageConsumerUUID, err := s.consumerManager.OnboardClient(ctx, consumerName, req, int(storageQuotaInGiB))
 	if err != nil {
-		if !kerrors.IsAlreadyExists(err) && err != errTicketAlreadyExists {
-			return nil, status.Errorf(codes.Internal, "failed to create storageConsumer %q. %v", req.ConsumerName, err)
-		}
-
-		storageConsumer, err := s.consumerManager.GetByName(ctx, req.ConsumerName)
-		if err != nil {
-			return nil, status.Errorf(codes.Internal, "Failed to get storageConsumer. %v", err)
-		}
-
-		if storageConsumer.Spec.Enable {
-			err = fmt.Errorf("storageconsumers.ocs.openshift.io %s already exists", req.ConsumerName)
-			return nil, status.Errorf(codes.AlreadyExists, "failed to create storageConsumer %q. %v", req.ConsumerName, err)
-		}
-		storageConsumerUUID = string(storageConsumer.UID)
+		return nil, status.Errorf(codes.Internal, "failed to onboard on storageConsumer resource. %v", err)
+	}
+	storageConsumer, err := s.consumerManager.Get(ctx, storageConsumerUUID)
+	if err != nil {
+		return nil, status.Errorf(codes.Internal, "failed to get storageConsumer. %v", err)
+	}
+	if storageConsumer.Spec.Enable {
+		err = fmt.Errorf("storageconsumers.ocs.openshift.io %s already exists", storageConsumer.Name)
+		return nil, status.Errorf(codes.AlreadyExists, "failed to onboard on storageConsumer resources. %v", err)
```

Comment:
Since we are not creating the consumer anymore in this rpc, could we update the err message to be something more meaningful like "storageconsumer already has a client attached to it"?